### PR TITLE
tony's submission

### PR DESCRIPTION
**Pull request by tonysun9**
Summary:
* Val loss: 3.498
* Wandb link: https://api.wandb.ai/links/tonysun9/sihutgml
* Dataset size: around 8B tokens

Steps:
1. Apply gopher filters at a document-level
2. Filter out documents with high duplicate content (below 50% unique lines or 5-grams)
3. Apply a subset of the filters at a line-level (aimed to filter out headers / navigation links)
4. English language ID filter (0.95 threshold)
5. Exact line deduplication (done within a WARC file for simplicity)

General approach:
* Estimate number of tokens
  * 128 batch size * 2 GPUs * 200k steps * 512 context size = 26B tokens
  * If training on the same tokens up to 4 times is okay, aim to create a dataset somewhere between 6.5B and 26B tokens
* Roughly try follow Dolma approach
  * Data preprocessing: duplicate content / line-level filters to remove headers and navigation links
  * Gopher quality filters.
  * Deduplication (arguably not necessary for this task looking at Figure 5 of the Dolma paper. Also did not add content filters for this reason)
  * English filter to match validation set

Other notes:
* Probably would've been smarter to apply language ID filter second. Follows principle of applying all document filters first, then all line filters. And order by speed within each category.

### CODE REVIEW

**Comment by nelson-liu** [comment]
thanks!
